frontend/src/component/layout/MainLayout/NavigationSidebar/NavigationList.tsx

```diff
@@ -245,6 +247,10 @@ export const RecentProjectsNavigation: FC<{
     projectId: string;
     onClick: () => void;
 }> = ({ mode, onClick, projectId }) => {
+    const { project, loading } = useProjectOverview(projectId);
+    const projectDeleted = !project.name && !loading;
+
+    if (projectDeleted) return null;
     return (
```

Comment:
Is there a specific benefit to doing this here rather than where `RecentProjectsNavigation` is referenced? (ie [here](https://github.com/Unleash/unleash/blob/main/frontend/src/component/layout/MainLayout/NavigationSidebar/NavigationSidebar.tsx#L62))

Reply:
yes, originally I tried to put it one component up the hierarchy. However this component should be conditional. Only when you have projectId should you call useProjectOverview. So by putting useProjectOverview here we make this hook conditional and the condition is one layer up.

Reply:
Sounds good! 👍 :shipit: